### PR TITLE
Don't intersect when clamping for size oblivious

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123675

Fixes https://github.com/pytorch/pytorch/issues/123651

Previously, when we performed a size oblivious test, we would only modify the lower bound, e.g., if we knew something had range `[0, 100]`, the size oblivious test would do `[2, 100]`. But what if your original range was `[0, 1]`? Naively intersecting this with `[2, sympy.oo]` would result in an empty set: that's a big no no. And in general, this intersection is kind of questionable: if your original range was `[0, 2]`, do we really want to assume that this quantity is exactly equal to 2 in the size oblivious test?

So here's an idea: when we're doing a size oblivious test, just forget about the max bound entirely. The idea is that the max bound probably wasn't actually helping you discharge the size oblivious test (because size oblivious tests are all about "well, if we can assume thing isn't zero or one, we know what the static value is.") So you can use the max bound OR you can use the size oblivious bound, but you're not allowed to use both at the same time. (It doesn't actually seem necessary to use the max bound, but it would be easy to permit this without using the size oblivious refinement.)

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Comment by ezyang** [comment]
Looks like this is OK to do. Will turn into a real PR tomorrow.

**Comment by ezyang** [comment]
updated and now with a test

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -1413,6 +1413,15 @@ def f(x):
             return r.view(12, -1, 192)
         make_fx(f, tracing_mode="symbolic")(torch.tensor(24))
 
+    def test_view_divisibility_unbacked_relatively_prime(self):
+        # See https://github.com/pytorch/pytorch/issues/123651
+        def f(x):
+            i0 = x.item()
+            torch._check_is_size(i0)
+            torch._check(i0 <= 1000000)
```
Comment (PaulZhang12) [inline test/test_proxy_tensor.py:1421]: Why was this changed from 448? Other than that, lgtm

**Review by PaulZhang12 — APPROVED** [review]

File: test/test_proxy_tensor.py (hunk 1 above)
Comment (ezyang) [reply test/test_proxy_tensor.py:1421]: Oh oops, I changed this for debugging and forgot to change it back
